node/sector_builder_submodule.go

```diff
@@ -0,0 +1,9 @@
+package node
+
+import "github.com/filecoin-project/go-filecoin/proofs/sectorbuilder"
+
+// SectorBuilderSubmodule enhances the `Node` with sector storage capabilities.
+type SectorBuilderSubmodule struct {
+	// SectorBuilder is used by the miner to fill and seal sectors.
+	sectorBuilder sectorbuilder.SectorBuilder
+}
```

Comment:
This should probably go with the storage protocol, though I'm not 100% sure about that.

Reply:
I left it out of it because it will be required by the retrieval protocol too. 
It is also possible that we might have a node running the storage protocol but not being responsible for actually storing the data locally, in which case they might have a different implementation of this submodule which talks with sector builder nodes.

Reply:
Right we need to split the StorageProtocol up. It must be possible to have a storage miner (committing sectors, submitting PoSt) without any deal protocol. The latter is built on the former.